great_expectations/datasource/new_datasource.py

```diff
@@ -261,6 +261,45 @@ def get_available_data_asset_names(
 
         return available_data_asset_names
 
+    def get_available_data_asset_names_and_types(
+        self, data_connector_names: Optional[Union[list, str]] = None
+    ) -> Dict[str, List[Tuple[str, str]]]:
+        """
+        Returns a dictionary of data_asset_names that the specified data
+        connector can provide. Note that some data_connectors may not be
+        capable of describing specific named data assets, and some (such as
+        inferred_asset_data_connector) require the user to configure
+        data asset names.
```

Comment:
I may be reading this incorrectly, but the one that requires the user to configure the data_asset names = `configured_asset_data_connector` ?

Reply:
hmm, i ported that over from the equivalent method without types. but you make a good point, it's possible that it's wrong there as well!